packages/gafl-webapp-service/src/pages/summary/licence-summary/__tests__/licence-summary.spec.js

```diff
@@ -220,4 +238,70 @@ describe('The licence summary page', () => {
     const data = await injectWithCookies('GET', LICENCE_SUMMARY.uri)
     expect(data.statusCode).toBe(200)
   })
+
+  it('date of birth amendment causes redirect the licence summary page', async () => {
+    const data = await injectWithCookies('GET', DATE_OF_BIRTH.uri)
+    expect(data.payload.search(backLinkRegEx(LICENCE_SUMMARY.uri)) > 0).toBeTruthy()
+    const data2 = await postRedirectGet(DATE_OF_BIRTH.uri, dobHelper(dobAdultToday))
+    expect(data2.statusCode).toBe(302)
+    expect(data2.headers.location).toBe(LICENCE_SUMMARY.uri)
+  })
+
+  it('date of birth (senior) amendment causes redirect the licence summary page and then the contact summary page', async () => {
+    await injectWithCookies('POST', DATE_OF_BIRTH.uri, dobHelper(dobSeniorToday))
+    const data = await injectWithCookies('GET', CONTROLLER.uri)
+    expect(data.statusCode).toBe(302)
+    expect(data.headers.location).toBe(LICENCE_SUMMARY.uri)
+    await injectWithCookies('GET', LICENCE_SUMMARY.uri)
+    await injectWithCookies('POST', LICENCE_SUMMARY.uri)
+
+    const data2 = await injectWithCookies('GET', CONTROLLER.uri)
+    expect(data2.statusCode).toBe(302)
+    expect(data2.headers.location).toBe(NAME.uri)
+  })
+
+  it('date of birth amendment (no licence required) causes a redirect to the no licence required page', async () => {
+    await postRedirectGet(LICENCE_TO_START.uri, { 'licence-to-start': 'after-payment' })
+    const data = await postRedirectGet(DATE_OF_BIRTH.uri, dobHelper(dobJuniorTomorrow))
+    expect(data.statusCode).toBe(302)
+    expect(data.headers.location).toBe(NO_LICENCE_REQUIRED.uri)
+    await injectWithCookies('GET', LICENCE_SUMMARY.uri)
+    await injectWithCookies('POST', LICENCE_SUMMARY.uri)
+
+    const data2 = await injectWithCookies('GET', CONTROLLER.uri)
+    expect(data2.statusCode).toBe(302)
+    expect(data2.headers.location).toBe(NAME.uri)
+  })
+
+  it('date of birth amendment (junior) causes redirect to the junior licence page and subsequent redirect to the licence summary page and then the contact summary page', async () => {
+    await injectWithCookies('POST', DATE_OF_BIRTH.uri, dobHelper(dobJuniorToday))
+    let data = await injectWithCookies('GET', CONTROLLER.uri)
+    expect(data.statusCode).toBe(302)
+    expect(data.headers.location).toBe(JUNIOR_LICENCE.uri)
+    data = await postRedirectGet(JUNIOR_LICENCE.uri, {})
+    expect(data.statusCode).toBe(302)
+    expect(data.headers.location).toBe(LICENCE_SUMMARY.uri)
```

Comment:
Could the further assertions after these be broken out to separate tests? We look to have achieved the objective of the test here (redirected to licence summary page and then contact summary page)

Reply:
ok - done